test/XRP/fakes/fake-xrp-client.ts

```diff
@@ -18,6 +18,7 @@ class FakeXrpClient implements XrpClientDecorator {
     public getPaymentValue: Result<XrpTransaction>,
     public enableDepositAuthValue: Result<TransactionResult>,
     public authorizeSendingAccountValue: Result<TransactionResult>,
+    public unauthorizeSendingAccountValue: Result<TransactionResult>,
```

Comment:
in https://github.com/xpring-eng/Xpring-JS/pull/757 we discussed just having a `this.successfulTransaction`, but that wouldn't quite work here - it would mean that we would need to explicitly call `this.successfulTransaction` in our functions, which would remove the benefits of DI. i know the constructor sux but seems like this is ok for now. 

in the future i propose we update this constructor to take an object where the defaults for each of the functions is a lambda that throws so that callers can only DI the things they care about

Reply:
oh I see what you mean... it would remove our ability to alter the individual return values in the case that we don't want them all to be successful, thus defeating the purpose of direct injection (DI?) .